BuildaGitServer/Base/Authentication.swift

```diff
@@ -69,4 +73,16 @@ extension ProjectAuthenticator: KeychainStringSerializable {
             self.secret
             ].joinWithSeparator(":")
     }
+
+    internal static func createEnterpriseService(host: String) -> GitService? {
+        guard let url = NSURL.init(string: "http://\(host)") else { return nil }
+        do {
+            let response = try NSString.init(contentsOfURL: url, encoding: NSASCIIStringEncoding)
+            if response.lowercaseString.containsString("github") {
```

Comment:
I could do this instead, but I don't think it's better because if someone with more permissions tries to use buildasaur it'll fail. :

    internal static func createEnterpriseService(host: String) -> GitService? {
        guard let url = NSURL(string: "http://\(host)/setup/api/settings")
            else { return nil }
        do {
            let response = try NSString(contentsOfURL: url, encoding: NSASCIIStringEncoding)
            if response.lowercaseString.containsString("not authorized") {
                return GitService.EnterpriseGitHub(host: host)
            }
        } catch {
        }
        return nil
    }